db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -474,6 +500,7 @@ Status DBImpl::AtomicFlushMemTablesToOutputFiles(
     s = error_status.ok() ? s : error_status;
   }
 
+  IOStatus io_s = IOStatus::OK();
```

Comment:
Maybe I am missing something.
If we have line 503 setting io_s to OK, then line 504 will always evaluate to true. Is this intended? @zhichao-cao ?